allensdk/brain_observatory/behavior/behavior_ophys_session.py

```diff
@@ -305,6 +310,88 @@ def segmentation_mask_image(self) -> Image:
+        - All positions are in units of pixels.
+        - All areas are in units of pixels^2
+        - All values are in the coordinate space of the eye tracking camera,
+          NOT the coordinate space of the stimulus display (i.e. this is not
+          gaze location), with (0, 0) being the upper-left corner of the
+          eye-tracking image.
+        - The 'likely_blink' column is True for any row (frame) where the pupil
+          fit failed OR eye fit failed OR an outlier fit was identified.
+        - All ellipse fits are derived from tracking points that were output by
+          a DeepLabCut model that was trained on hand-annotated data frome a
+          subset of imaging sessions on optical physiology rigs.
+        - Raw DeepLabCut tracking points are not publicly available.
+
+        :rtype: pandas.DataFrame
+        """
+        kwargs = self.eye_tracking_processing_args
+
+        logging.getLogger("BehaviorOphysSession").info(
+            f"Eye tracking data retrieved with the following parameters:"
+            f" {kwargs}"
+        )
+
+        if self._eye_tracking_data is None:
+            self._eye_tracking_data = self.api.get_eye_tracking_data(**kwargs)
+        return self._eye_tracking_data
+
+    @eye_tracking_data.setter
+    def eye_tracking_data(self, value):
+        self._eye_tracking_data = value
+
+    @property
+    def eye_tracking_processing_args(self) -> dict:
+        """Arguments used when processing 'eye_tracking_data'.
+
+        This must be a python dictionary with the following items:
+
+        z_threshold : float
+            z-score values higher than the z_threshold in the
+            'eye_tracking_data' will be considered outliers, by default 3.0.
+
+        dilation_frames : int
+            Determines the number of additional adjacent frames to mark as
+            'likely_blink' when detecting likely blinks for the
+            'eye_tracking_data', by default 2.
+
+        :rtype: dict
+        """
+        if self._eye_tracking_processing_args is None:
+            self._eye_tracking_processing_args = {"z_threshold": 3.0,
+                                                  "dilation_frames": 2}
+        return self._eye_tracking_processing_args
+
+    @eye_tracking_processing_args.setter
+    def eye_tracking_processing_args(self, value: dict):
+        if ((not isinstance(value, dict)) or ("z_threshold" not in value) or ("dilation_frames" not in value)):
+            raise TypeError("'eye_tracking_processing_args' must be a dict "
+                            "with the following items: {'z_threshold': "
+                            "(float), 'dilation_frames': (int)}")
+
+        self._eye_tracking_processing_args = value
+        logging.getLogger("BehaviorOphysSession").info(
+            "'eye_tracking_processing_args' have been changed! "
+            "The session 'eye_tracking_data' will now be updated."
+        )
+        _ = self.eye_tracking_data
```

Comment:
`self._eye_tracking_data = None`

Reply:
Yeah, I forgot to commit this change. It's in now.

Reply:
Since that's what the eye tracking property is checking to see if it needs to pull data from the api:
```
if self._eye_tracking_data is None:
        self._eye_tracking_data = self.api.get_eye_tracking_data(**kwargs)
        return self._eye_tracking_data
```